core/frontend/src/tile/RealityModelTileTree.ts

```diff
@@ -711,7 +713,7 @@ export namespace RealityModelTileTree {
       const props = await getTileTreeProps(rdSource, tilesetToDb, iModel);
       const loader = new RealityModelTileLoader(props, new BatchedTileIdMap(iModel), opts);
       const gcsConverterAvailable = await getGcsConverterAvailable(iModel);
-      const params = new RealityModelTileTreeParams(tileTreeId, iModel, modelId, loader, gcsConverterAvailable, props.tilesetToEcef);
+      const params = new RealityModelTileTreeParams(tileTreeId, iModel, modelId, loader, gcsConverterAvailable, props.tilesetToEcef, rdSource.key.id);

```

Comment:
I was interpreting it as a base url due to its creation in `createKeyFromUrl`, [here](https://github.com/iTwin/itwinjs-core/blob/7bdb8cb11bbbe9a3f7fe0c9eb0d3f7191ba48934/core/frontend/src/RealityDataSource.ts#L131), where the function defaults to setting the reality data source key id as the tileset url. The thinking was that within the other two paths in that function, the id wouldn't be a valid url, so then when we check for a valid url later, the ids wouldn't be added to the reader, so no issue. 

After looking into this more, this does leave the possibility that the source id is created using one of the other paths in `createKeyFromUrl`, meaning the id might not be a valid url, but there still could be relative texture uris within the tileset, meaning we still need to pass along a base url. Changes are underway now to pass along the tileset url used to create the reality data source key instead of just the key id itself, thereby avoiding these issues.

Reply:
Did you notice that RealityDataSourceTilesetUrlImpl already has a base URL?

Reply:
Thanks for the suggestions on where to look, a few things have now been updated - `getTilesetUrlFromTilesetUrlImpl` has been added to `RealityDataSource` which obtains the tileset url from a source which is an instance of `RealityDataSourceTilesetUrlImpl`. This is being used to set the `baseUrl` property in `RealityModelTileTree`. A note: we need to get the full tileset url, not just the base url, from `RealityDataSourceTilesetUrlImpl`. This is because `ResolveUrl` in `GltfReader`, [here](https://github.com/iTwin/itwinjs-core/blob/4a92d6853949bada726a3d14ec29b0c1b85a7622/core/frontend/src/tile/GltfReader.ts#L1958), expects the base url to have search parameters if they are necessary, and if we were to only use the base url from `RealityDataSourceTilesetUrlImpl` for something like a tileset in blob storage, these search parameters would not be present, and the texture requests would return forbidden.